src/graphql/resolvers/queries/asset.js

```diff
@@ -17,9 +17,28 @@ export const fieldResolvers = {
       return assetLoader.load(id);
     },
 
-    assignments: ({ id: assetId }, __, { dataSources: { firestore } }) => {
+    assignments: async (
+      { id: assetId },
+      __,
+      { dataSources: { firestore, sessionLoader } },
+    ) => {
       dlog(`asset's assignments %s`, assetId);
-      return assetStore(firestore).getAssetAssignments(assetId);
+      const assignments = await assetStore(firestore).getAssetAssignments(
+        assetId,
+      );
+      const promises = assignments.map(a => {
+        if (a.entityType === 'SESSION') {
+          return sessionLoader
+            .load(a.id)
+            .then(session => ({ ...session, entityType: a.entityType }));
+        }
+        return a;
+      });
+      return Promise.all(promises);
     },
+
+    createdBy: ({ createdBy }) => ({ id: createdBy }),
```

Comment:
what's going on with createdBy and lastUpdatedBy?

Reply:
This is returning an object of { id: `memberId` } so that a PublicProfile record can be fetched by federation. 

`createdBy` and `lastUpdatedBy` hold a member id value. That id needs to be in the form of an object so it can be resolved correctly. 

Is that clear?